lib/bolt_server/transport_app.rb

```diff
@@ -280,17 +295,25 @@ def make_winrm_target(target_hash)
       [200, result_set_to_status_hash(result_set, aggregate: aggregate).to_json]
     end
 
+    get '/plans/:module_name/:plan_name' do
+      in_pe_pal_env(params['environment']) do |pal|
+        plan_info = pal.get_plan_info("#{params[:module_name]}::#{params[:plan_name]}")
+        [200, plan_info.to_json]
+      end
+    end
+
     get '/plans' do
-      @pal_mutex.synchronize do
-        environment = params['environment']
-        if environment.nil?
-          [400, '`environment` is a required argument']
-        else
-          pal = BoltServer::PE::PAL.new({}, params['environment'] || 'production')
-          plans = pal.list_plans.flatten
-          plan_info = plans.each_with_object({}) { |plan_name, acc| acc[plan_name] = pal.get_plan_info(plan_name) }
-          [200, plan_info.to_json]
-        end
+      in_pe_pal_env(params['environment']) do |pal|
+        plans = pal.list_plans.flatten.map { |plan| { 'name' => plan } }
+        [200, plans.to_json]
+      end
+    end
+
+    get '/plans_metadata' do
```

Comment:
I wonder if this and the `/plans` endpoint can be merged with a `?detail` argument defaulting to `true` or something?

Reply:
Oh, yeah that might be a good idea

Reply:
Actually `metadata` might be a better parameter name.